agent/check_handler.go

```diff
@@ -223,3 +224,44 @@ func extractMetrics(event *types.Event) []*types.MetricPoint {
 
 	return transformer.Transform()
 }
+
+// mergeEnvironments merges env2 into env1, overwriting any existing variable
+// in env1, except for the "special" variables PATH, CPATH and LD_LIBRARY_PATH.
+// These variables are merged by prepending the value from env2 to the value in
+// env1, effectively giving priority to the value from env2.
+func mergeEnvironments(env1, env2 []string) []string {
+	toMap := func(s []string) map[string]string {
```

Comment:
Defining this in the scope of mergeEnvironments is a bit odd. It captures the env1 and env2 parameters, which are not used in the function. Why not move it out to the package scope?

Reply:
What was odd to me was to represent the environment with `[]string` :)

Here was my reasoning for not defining `toMap()` and `fromMap()` in the package scope: they are fairly generic, but they have to be restricted to `[]string -> map[string]string` and `map[string]string -> []string` respectively in Go. So they become one off helpers as far as I'm concerned, because I don't want to pollute the package scope with either a function name that imply genericity or a function name that looks like `fromStringSliceToMapStringString()`.

You can probably guess I wasn't in a very Go mood when I wrote that code! I'm happy to go with whatever is "idiomatic Go" though.

Reply:
```
¯\_(ツ)_/¯
```